builder/simplify.cc

```diff
@@ -2060,10 +2060,52 @@ class simplifier : public node_mutator {
     visit_crop(src == op->src ? op : nullptr, op->sym, src, std::move(bounds), op->body);
   }
 
+  bool prove_contains(interval_expr outer, interval_expr inner) {
+    return prove_true(outer.contains(inner.min)) && prove_true(outer.contains(inner.max));
+  }
+
+  bool prove_slice_unaffected(const std::vector<interval_expr>& bounds, const std::vector<expr>& at) {
+    for (size_t dim = at.size(); dim < bounds.size(); ++dim) {
+      if (!prove_contains(bounds[dim], interval_expr::all())) {
```

Comment:
I think it should be sufficient here to just say `if (bounds.min.defined() || bounds.max.defined()) return false`.